packages/opencensus-core/src/tags/validation.ts

```diff
@@ -0,0 +1,42 @@
+/**
+ * Copyright 2019, OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 the "License";
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {TagKey, TagValue} from './types';
+
+const nonPrintableCharsRegex = /[^\u0020-\u007e]/;
+const TAG_KEY_MAX_LENGTH = 255;
+
+/** Determines whether the given String is a valid tag key. */
+export function isValidTagKey(tagKey: TagKey): boolean {
+  if (!tagKey) {
+    return false;
+  }
+  return isPrintableString(tagKey.name) && tagKey.name.length > 0 &&
+      tagKey.name.length <= TAG_KEY_MAX_LENGTH;
+}
+
+/** Determines whether the given String is a valid tag value. */
+export function isValidTagValue(tagValue: TagValue): boolean {
+  if (!tagValue) {
+    return false;
+  }
+  return isPrintableString(tagValue.value) &&
+      tagValue.value.length <= TAG_KEY_MAX_LENGTH;
```

Comment:
`TagValue` doesn't have a length constraint (https://github.com/census-instrumentation/opencensus-specs/blob/master/tags/TagMap.md#tagvalue).

Reply:
I have compared with [Java](https://github.com/census-instrumentation/opencensus-java/blob/master/api/src/main/java/io/opencensus/tags/TagValue.java#L77) and [Python](https://github.com/census-instrumentation/opencensus-python/blob/master/opencensus/tags/validation.py#L42), I see we have enforced length constaint on ```TagValue```. Let me know if you still want me to remove it.

Reply:
I see. We can leave the constraint for now, it's easy to relax it in the future.